app/models/ability.rb

```diff
@@ -0,0 +1,14 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user
+
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :read, :all
+      can :delete, [Post, Comment], author_id: user.id
```

Comment:
- [ ] Right now, I can't create a post as a logged in user. Please fix it. 
![image](https://github.com/katfogy/blogApp/assets/91486791/5b398dea-0fde-475d-a7fe-f09e19d2286a)

Reply:
Hi @tugsem is creating, is only admin can create, and not normal user, please check

Reply:
@katfogy A user can delete a post or comment if it belongs to that user. Otherwise, if the user is 'admin', he can also manage, and delete. But as you have set in the ability.rb, the post must belong to that user's id for a user to delete his post, not just belong to an admin. Happy coding!